Mathlib/NumberTheory/Padics/PadicIntegers.lean

```diff
@@ -258,6 +258,22 @@ theorem norm_int_le_pow_iff_dvd {k : ℤ} {n : ℕ} :
     simpa [norm_intCast_eq_padic_norm]
   padicNormE.norm_int_le_pow_iff_dvd _ _
 
+lemma norm_natCast_eq_one_iff {n : ℕ} :
+    ‖(n : ℤ_[p])‖ = 1 ↔ p.Coprime n := by
+  rw [norm_def, coe_natCast, Padic.norm_natCast_eq_one_iff]
+
+lemma norm_natCast_lt_one_iff {n : ℕ} :
+    ‖(n : ℤ_[p])‖ < 1 ↔ p ∣ n := by
+  rw [norm_def, coe_natCast, Padic.norm_natCast_lt_one_iff]
+
+lemma norm_intCast_eq_one_iff {z : ℤ} :
+    ‖(z : ℤ_[p])‖ = 1 ↔ IsCoprime z p := by
+  rw [norm_def, coe_intCast, Padic.norm_intCast_eq_one_iff]
+
+lemma norm_intCast_lt_one_iff {z : ℤ} :
+    ‖(z : ℤ_[p])‖ < 1 ↔ (p : ℤ) ∣ z := by
+  rw [norm_def, coe_intCast, Padic.norm_intCast_lt_one_iff]
```

Comment:
Should those all be simp?

Reply:
Sure, but the LHS mentions a complicated objects, the p-adics, while the RHS is only about `Nat`/`Int`. 

> if the RHS is dvd, there aren't really that many further simp lemmas that would apply.

How so? There are plenty of lemmas about `dvd`, and you can use `norm_num` if they are numerals too

Reply:
Added simp tags